File: src/main/java/org/funivan/lologa/tile/Score/ScoreSum.java
package org.funivan.lologa.tile.Score;

import org.funivan.lologa.tile.TileInterface;
import org.funivan.lologa.tiles.TilesInterface;

public class ScoreSum implements ScoreInterface {
    private final Iterable<TileInterface> tiles;

    public ScoreSum(Iterable<TileInterface> tiles) {
        this.tiles = tiles;
    }
    public ScoreSum(TilesInterface tiles) {
        this.tiles = tiles.all();
    }

    @Override
    public int value() {
        int score = 0;
        for (TileInterface tile : this.tiles) {
            score = score + tile.score().value();
        }
        return score;
    }

    @Override
    public String toString() {
        return "Score{" + this.value() + '}';
    }
}

File: src/main/java/org/funivan/lologa/tile/Score/ScoreSum.kt
package org.funivan.lologa.tile.Score

import org.funivan.lologa.tile.TileInterface
import org.funivan.lologa.tiles.TilesInterface

class ScoreSum : ScoreInterface {
    private val tiles: Iterable<TileInterface>

    constructor(tiles: Iterable<TileInterface>) {
        this.tiles = tiles
    }

    constructor(tiles: TilesInterface) {
        this.tiles = tiles.all()
    }

    override fun value(): Int {
        var score = 0
        for (tile in this.tiles) {
            score = score + tile.score().value()
        }
        return score
    }

    override fun toString(): String {
        return "Score{" + this.value() + '}'
    }
}
